gtsam/hybrid/tests/testGaussianMixture.cpp

```diff
@@ -78,15 +78,51 @@ TEST(GaussianMixture, Equals) {
   GaussianMixture::Conditionals conditionals(
       {m1},
       vector<GaussianConditional::shared_ptr>{conditional0, conditional1});
-  GaussianMixture mixtureFactor({X(1)}, {X(2)}, {m1}, conditionals);
+  GaussianMixture mixture({X(1)}, {X(2)}, {m1}, conditionals);
 
   // Let's check that this worked:
   DiscreteValues mode;
   mode[m1.first] = 1;
-  auto actual = mixtureFactor(mode);
+  auto actual = mixture(mode);
   EXPECT(actual == conditional1);
 }
 
+/* ************************************************************************* */
+/// Test error method of GaussianMixture.
+TEST(GaussianMixture, Error) {
+  Matrix22 S1 = Matrix22::Identity();
+  Matrix22 S2 = Matrix22::Identity() * 2;
+  Matrix22 R1 = Matrix22::Ones();
+  Matrix22 R2 = Matrix22::Ones();
+  Vector2 d1(1, 2), d2(2, 1);
+
+  SharedDiagonal model = noiseModel::Diagonal::Sigmas(Vector2(1.0, 0.34));
+
+  auto conditional0 = boost::make_shared<GaussianConditional>(X(1), d1, R1,
+                                                              X(2), S1, model),
+       conditional1 = boost::make_shared<GaussianConditional>(X(1), d2, R2,
+                                                              X(2), S2, model);
+
+  // Create decision tree
+  DiscreteKey m1(1, 2);
+  GaussianMixture::Conditionals conditionals(
+      {m1},
+      vector<GaussianConditional::shared_ptr>{conditional0, conditional1});
+  GaussianMixture mixture({X(1)}, {X(2)}, {m1}, conditionals);
+
+  VectorValues values;
+  values.insert(X(1), Vector2::Ones());
+  values.insert(X(2), Vector2::Zero());
+  auto error_tree = mixture.error(values);
+
+  std::vector<DiscreteKey> discrete_keys = {m1};
+  std::vector<double> leaves = {0.5, 4.3252595};
```

Comment:
Always comment on numerical values in tests, either "regression" (you copied from output and it's not a real test), or "calculated by hand" or something else.

Reply:
Ah, I added "regression" to the `EXPECT` call right below. Let me move it up.